src/core/src/constant_fold_utils.cpp

```diff
@@ -0,0 +1,279 @@
+#include "openvino/pass/constant_folding.hpp"
+#include "openvino/reference/convert.hpp"
+#include "ov_ops/type_relaxed.hpp"
+
+ov::element::TypeVector ov::util::unsupported_types() {
+    return {ov::element::f16, ov::element::bf16};
+}
+
+bool ov::util::node_requires_precision_conversion(const std::shared_ptr<const ov::Node>& node) {
+    if (node->get_input_size() == 0)
+        return false;
+    if (node->has_evaluate()) {
+        return false;
+    }
+    // ReadValue or Assign cannot be cloned with new inputs with different precision
+    if (ov::is_type<ov::op::util::ReadValueBase>(node) || ov::is_type<ov::op::util::AssignBase>(node)) {
+        return false;
+    }
+    // ConvertLike has constant_fold function but doesn't have has_evaluate function
+    if (ov::is_type<ov::op::v1::ConvertLike>(node)) {
+        return false;
+    }
+    if (ov::is_type<ov::op::util::MultiSubGraphOp>(node)) {
+        return false;
+    }
+    const auto unsupported_types = ov::util::unsupported_types();
+    for (size_t i = 0; i < node->get_input_size(); i++) {
+        if (std::find(unsupported_types.begin(), unsupported_types.end(), node->get_input_element_type(i)) !=
+            unsupported_types.end()) {
+            return true;
+        }
+    }
+    for (size_t i = 0; i < node->get_output_size(); i++) {
+        if (std::find(unsupported_types.begin(), unsupported_types.end(), node->get_output_element_type(i)) !=
+            unsupported_types.end()) {
+            return true;
+        }
+    }
+    return false;
+}
+
+static bool convert_range_precision(const std::shared_ptr<ov::Node>& node) {
+    auto range = ov::as_type_ptr<ov::op::v4::Range>(node);
+    if (!range)
+        return false;
+
+    const auto unsupported_types = ov::util::unsupported_types();
+    if (std::find(unsupported_types.begin(), unsupported_types.end(), range->get_output_type()) !=
+        unsupported_types.end()) {
+        range->set_output_type(ov::element::f32);
+        return true;
+    }
+    return false;
+}
+
+static std::unordered_map<ov::NodeTypeInfo, std::function<bool(const std::shared_ptr<ov::Node>&)>>
+    output_conversion_methods = {
+        {ov::op::v4::Range::get_type_info_static(), convert_range_precision},
+};
+
+std::shared_ptr<ov::Node> ov::util::convert_to_supported_precision(const std::shared_ptr<ov::Node>& node) {
+    return ov::util::convert_to_supported_precision(node, node->input_values());
+}
+
+std::shared_ptr<ov::Node> ov::util::convert_to_supported_precision(const std::shared_ptr<ov::Node>& node,
+                                                                   OutputVector&& inputs) {
+    size_t num_inputs = node->get_input_size();
+    if (num_inputs == 0)
+        return node;
+
+    if (ov::is_type<ov::op::util::MultiSubGraphOp>(node))
+        return node;
+
+    bool requires_conversion = false;
+    bool inputs_changed = false;
+
+    const auto unsupported_types = ov::util::unsupported_types();
```

Comment:
```suggestion
    const auto& unsupported_types = ov::util::unsupported_types();
```
If functions returns reference to types vector, apply in all occurrences then